src/jdk.jfr/share/classes/jdk/jfr/internal/tracing/TimedClass.java

```diff
@@ -60,13 +61,23 @@ public void emit(long timestamp) {
                 long methodId = tm.method().methodId();
                 long invocations = tm.invocations().get();
                 long time = tm.time().get();
-                long average = invocations == 0 ? Long.MIN_VALUE : time / invocations;
                 long min = tm.minimum().get();
-                if (min == Long.MAX_VALUE) {
-                    min = Long.MIN_VALUE; // Signals that the value is missing
-                }
                 long max = tm.maximum().get();
-                MethodTimingEvent.commit(timestamp, methodId, invocations, min, average, max);
+                if (time == 0 || invocations == 0) {
+                    // If time is zero, it's a low resolution clock and more invocations are needed.
+                    MethodTimingEvent.commit(timestamp, methodId, invocations, MISSING, MISSING, MISSING);
+                } else {
+                    long average = (time + invocations / 2) / invocations;
+                    if (min == Long.MAX_VALUE) {
+                        min = average;
+                    }
+                    if (max == Long.MIN_VALUE) {
+                        max = average;
+                    }
```

Comment:
For the future: I think `Math.min/max` below subsume these checks for `Long.MAX/MIN_VALUE`.